backend/parser.py

```diff
@@ -49,42 +68,138 @@ def add_point(self, timestamp, data: list, latency=None):
         else:
             self.timestamp_diffs.append(0)
 
-        if latency != None:
+        if latency >= 0:
             self.latencies.append(latency)
 
-        for i, point in enumerate(data):
+        for i, point in enumerate(datapoints):
             self.data[i].append(point)
             
         self.next_index += 1
 
+    def set_dimensions(self, dimensions: int):
+        """Initializes empty lists in dict after parser determines data dimensions.
+        
+        Args:
+            dimensions: The number of dimensions that the data has. e.g. if the
+                gyroscope returns (x,y,z) tuples the parser calls set_dimensions(3).
+        
+        Raises:
+            ValueError: "<= 1 dimension error" 1 dimensions is always initialized and negative/0 dimensions don't exist.
+        """
+
+        if dimensions < 1:
+            raise ValueError("< 1 dimension error")
+        if dimensions == 1:
+            self.performed_dimension_set = True
+            return
+
+        for i in range(1, dimensions):
+            self.data[i] = []
+
+        self.performed_dimension_set = True
+
 class Parser:
     """Parser takes in one or more files and converts them to JSON
 
     Attributes:
         files: A list of files to parse
-        header: A regex that defines the header of the file, including fields such as sensor_name
-        body: A regex that defines the body of the file, including all relevant data
-            Header and body attributes allow extending file formats.
+        regex: A dict that holds the string regular expressions that match to various fields.
+            Keys:
+                'sensor_name': (Required) Matches the name of the sensor.
+                'sensor_id': (Required) Matches the ID of the sensor.
+                'timestamp': (Required) Matches every timestamp in the file.
+                'data': (Required) Matches the comma or space separated data fields. Should match all data points.
+                'latency': Matches all latency points.
```

Comment:
latency is currently missing from the regexes.

Is inline_id needed for something?

Reply:
inline_id is needed when multiple sensors are sampled in the same file. Each line is started with "Sensor: x" and inline_id matches the x so that the correct Sample can be updated.

Reply:
Got it, that makes sense.